### ISSUE TITLE
:Clap tags does not work if Vista haven't already open

### ISSUE THREAD

**Issue by mnrce** [liuchengxu/vim-clap#371]
**Environment (please complete the following information):**
 - OS: Ubuntu 16.04
 - (Neo)Vim version: NVIM v0.5.0-416-g2b00d1d09
 - vim-clap version: 8130b9bbcfc15be97184e08f1223ed2160417361
 - Have you reproduced with a minimal vimrc: Yes
 - Have you updated to the latest plugin version: Yes
 - Have you upgraded to/compiled the latest Rust binary: I install/update vim-clap with `  Plug 'liuchengxu/vim-clap', { 'do': ':call clap#installer#download_binary()' }`

**Describe the bug**
`:Clap tags` does not work if I don't run `:Vista!!` first.

**Clap debug**

```
            has cargo: 1
            has maple: /home/nghialm/.config/nvim/plugged/vim-clap/bin/maple
           maple info: version 0.1.9 (git v0.9), built for x86_64-unknown-linux-gnu by rustc 1.41.1 (f3e1a954d 2020-02-24).

         has +python3: 1
has py dynamic module: 0
     Current FileType: jsonc
Third Party Providers: []
       Global Options:
    let g:clap#autoload_dir = '/home/nghialm/.config/nvim/plugged/vim-clap/autoload'
    let g:clap#provider_alias = {'gfiles': 'git_files', 'hist:': 'command_history', 'hist/': 'search_history'}
    let g:clap_disable_bottom_top = 0
    let g:clap_enable_icon = 0
    let g:clap_forerunner_status_sign = {'done': '•', 'running': '!'}
    let g:clap_insert_mode_only = v:false
    let g:clap_no_matches_msg = 'NO MATCHES FOUND'
    let g:clap_open_action = {'ctrl-v': 'vsplit', 'ctrl-x': 'split', 'ctrl-t': 'tab split'}
    let g:clap_popup_border = 'rounded'
    let g:clap_search_box_border_style = 'nil'
    let g:clap_search_box_border_symbols = {'nil': ['', ''], 'curve': ['', ''], 'arrow': ['', '']}
  Provider Variables:
    let g:clap_providers_relaunch_code = '@@'
```

**To Reproduce**

1. Create the minimal vimrc `min.vim`:

```vim
let g:vista_default_executive = 'coc'
set nocompatible
set runtimepath^=/home/nghialm/.config/nvim/plugged/vim-jsonc,/home/nghialm/.config/nvim/plugged/coc.nvim,/home/nghialm/.config/nvim/plugged/vim-clap,/home/nghialm/.config/nvim/plugged/vista.vim
syntax on
filetype plugin indent on
```

2. Start (neo)vim with command: `nvim -u ~/min.vim tsconfig.json`

3. Wait for a second for coc.nvim to load.

4. Run `:Clap tags`

5. See error.

6. If I run `:Vista!!` before running `:Clap tags` then vim-clap works fine.

**Expected behavior**
vim-clap works fine and show a list of available symbols.

**Screenshots**
![image](https://user-images.githubusercontent.com/13317164/78244294-7e2c1980-750f-11ea-99ab-2fc8c440f851.png)
![image](https://user-images.githubusercontent.com/13317164/78244307-84ba9100-750f-11ea-8d08-73e940864a1c.png)

**GIF**
![a](https://user-images.githubusercontent.com/13317164/78244817-8fc1f100-7510-11ea-936b-70ddb4782043.gif)

**Additional context**
It doesn't work on every filetype, not only jsonc

**Comment by mnrce** [comment]
@liuchengxu: Thanks for the quick fix, I no longer get the error, but the `:Clap tags` only shows the first symbol if I haven't already run `:Vista!!`.

If I run `:Vista!!` before running `:Clap tags`, then all the symbols are shown.

And, `:Vista!!` no longer works after I run `:Clap tags`



**GIF**:
![a](https://user-images.githubusercontent.com/13317164/78248953-bd5e6880-7517-11ea-9f6a-249846d3f666.gif)

**Comment by liuchengxu** [comment]
@mnrce I can't reproduce that problem. Can you try scolling the result list to see if there is only one item or the window is wrongly set to one row height.

**Comment by mnrce** [comment]
> @mnrce I can't reproduce that problem. Can you try scolling the result list to see if there is only one item or the window is wrongly set to one row height.

The top-right of Clap popup shows `[1]`, so there should be one result.


> I also notice that when there is only one symbol, the symbol provider is ctags, when there are more symbols, the symbols provider is coc, so it probabaly is expected, as the different providers can provide different symbols.

Oh, I just notice that now. This is probably a vista.vim issue since I have `let g:vista_default_executive = 'coc'` at the beginning. Maybe I have to run `:Vista!!` first for that configuration to take effect?

**Comment by mnrce** [comment]
> > @mnrce I can't reproduce that problem. Can you try scolling the result list to see if there is only one item or the window is wrongly set to one row height.
> 
> The top-right of Clap popup shows `[1]`, so there should be one result.
> 
> > I also notice that when there is only one symbol, the symbol provider is ctags, when there are more symbols, the symbols provider is coc, so it probabaly is expected, as the different providers can provide different symbols.
> 
> Oh, I just notice that now. This is probably a vista.vim issue since I have `let g:vista_default_executive = 'coc'` at the beginning. Maybe I have to run `:Vista!!` first for that configuration to take effect?

> And, :Vista!! no longer works after I run :Clap tags

Are you able to reproduce this? As in the GIF above, after `:Clap tags` I ran `:Vista!!` but the vista window didn't show up.

**Comment by liuchengxu** [comment]
>This is probably a vista.vim issue 

That's a feature instead: vista.vim will fall back to the alternative executives when your default one does not produce any data. And coc is not always fast enough, also not stable enough, see this issue https://github.com/liuchengxu/vista.vim/issues/237#issuecomment-583859662. You are using coc as the default executive, which is a compicated plugin, plus the language server is not fast always, it can't stably produce the symbols from my experience.

>And, :Vista!! no longer works after I run :Clap tags

I don't see this, you should report to vista.vim if you can reproduce this.

**Comment by comfortablynick** [comment]
I’m seeing the same issue. In my minimal vimrc I’m not using coc, just opening the `minirc.vim` file and pulling up the one function using `:Clap tags`.

If I execute `:Clap` first, then scroll to `tags`, it works fine. But an initial command of `:Clap tags` brings up this error:

### Error
```
Error detected while processing function clap#[35]..clap#for[23]..<SNR>134_detect_source_type[9]..71[31]..clap#spinner#refresh[1]..<SNR>151_set_spinner[2]..clap#spinner#set[3]..64:
line    1:
E121: Undefined variable: s:spinner_winid
E116: Invalid arguments for function nvim_win_is_valid
E15: Invalid expression: nvim_win_is_valid(s:spinner_winid)
```

### Minimal vimrc
```vim
set runtimepath=$VIM/vimfiles,$VIMRUNTIME,$VIM/vimfiles/after
set runtimepath^=~/vim-test/pack/test/opt/vim-clap,~/vim-test/pack/test/opt/vista.vim
let &packpath=&runtimepath

filetype plugin indent on
syntax enable

function! Test() abort
    " Test function should show in :Clap tags
    return
endfunction
```

### Clap debug
```vim
            has cargo: 1
            has maple: v:null
         has +python3: 1
has py dynamic module: 0
     Current FileType: vim
Third Party Providers: []
       Global Options:
    let g:clap#autoload_dir = '/home/nick/vim-test/pack/test/opt/vim-clap/autoload'
    let g:clap#provider_alias = {'gfiles': 'git_files', 'hist:': 'command_history', 'hist/': 'search_history'}
    let g:clap_disable_bottom_top = 0
    let g:clap_enable_icon = 0
    let g:clap_forerunner_status_sign = {'done': '•', 'running': '!', 'using_cache': '*'}
    let g:clap_insert_mode_only = v:false
    let g:clap_no_matches_msg = 'NO MATCHES FOUND'
    let g:clap_open_action = {'ctrl-v': 'vsplit', 'ctrl-x': 'split', 'ctrl-t': 'tab split'}
    let g:clap_popup_border = 'rounded'
    let g:clap_search_box_border_style = 'nil'
    let g:clap_search_box_border_symbols = {'nil': ['', ''], 'curve': ['', ''], 'arrow': ['', '']}
  Provider Variables:
    let g:clap_providers_relaunch_code = '@@'
```

**Comment by liuchengxu** [comment]
@comfortablynick Please pull the latest master, should be fixed now.

**Comment by comfortablynick** [comment]
@liuchengxu thanks, it seems to have fixed nvim. When I try using the same minirc with vim, I get this error with the latest master:

```
Error detected while processing function clap#[35]..clap#for[23]..<SNR>31_detect_source_type[9]..53[31]..clap#spinner#refresh[1]..<SNR>48_set_spinner[2]..clap#spinner#set:
line    2:
E121: Undefined variable: g:clap_spinner_winid
E116: Invalid arguments for function popup_settext
```

**Comment by liuchengxu** [comment]
@comfortablynick Ok, fixed for vim in 894a6fb.

**Comment by comfortablynick** [comment]
@liuchengxu Thanks! Working in both vim and nvim now.
